File: server/src/db/entities/TimeSheetReport.ts
import { Entity, Column, OneToMany } from "typeorm";

import BaseEntity from "./BaseEntity";
import TimeSheetEntry from "./TimeSheetEntry";

type TimeSheetEntryStatus =
  | "IN_PROGRESS"
  | "COMPLETED"
  | "NEEDS_APPROVAL"
  | "PAID";

@Entity()
export default class TimeSheetReport extends BaseEntity {
  @Column({ type: "string", nullable: false, default: "PENDING" })
  status: TimeSheetEntryStatus;

  @Column({ type: "timestamptz" })
  payPeriodStart: Date;

  @Column({ type: "timestamptz" })
  payPeriodEnd: Date;

  // Relationships
  @OneToMany((type) => TimeSheetEntry, (t) => t.report)
  timeSheetEntries: TimeSheetEntry[];
}

File: Models/TimeSheetReport.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace TennisShopGuru.Models
{
  public enum TimeSheetReportStatus
  {
    ACTIVE,
    FINALIZED,
    PAID
  }

  public class TimeSheetReport : BaseEntity
  {
    public int Id { get; set; }
    public TimeSheetReportStatus Status { get; set; }
    public DateTime PayPeriodStart { get; set; }
    public DateTime PayPeriodEnd { get; set; }

    // Relationships
    public ICollection<TimeSheetEntry> TimeSheetEntries { get; set; }

    public int CompanyID { get; set; }
    [ForeignKey("CompanyID")]
    public Company Company { get; set; }
  }
}
